internal/buildengine/engine.go

```diff
@@ -863,14 +865,18 @@ func (e *Engine) BuildAndDeploy(ctx context.Context, replicas int32, waitForDepl
 	buildGroup.Go(func() error {
 		return e.buildWithCallback(ctx, func(buildCtx context.Context, module Module) error {
 			e.modulesToBuild.Store(module.Config.Module, false)
-			modulesToDeploy = append(modulesToDeploy, module)
 			e.rawEngineUpdates <- &buildenginepb.EngineEvent{
 				Event: &buildenginepb.EngineEvent_ModuleDeployWaiting{
 					ModuleDeployWaiting: &buildenginepb.ModuleDeployWaiting{
 						Module: module.Config.Module,
 					},
 				},
 			}
+			if singleChangeset {
```

Comment:
supernit: Ugly if statement we can put `return nil` in the if case and remove the `else`